src/main/java/hw03/CustomDynamicDeque.java

```diff
@@ -0,0 +1,219 @@
+package hw03;
+
+import hw01.CustomList;
+
+import java.util.Collection;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class CustomDynamicDeque<T> implements Deque<T> {
+    private final CustomList<T> list;
+
+    public CustomDynamicDeque() {
+        this.list = new CustomList<>();
+    }
+
+    @Override
+    public void addFirst(T t) {
+        list.add(0, t);
+    }
+
+    @Override
+    public void addLast(T t) {
+        list.add(size() - 1, t);
+    }
+
+    @Override
+    public boolean offerFirst(T t) {
+        addFirst(t);
+        return true;
+    }
+
+    @Override
+    public boolean offerLast(T t) {
+        return list.add(t);
+    }
+
+    @Override
+    public T removeFirst() {
+        return list.remove(0);
```

Comment:
would it be possible to write a dequeue backed by an array that would combine have quick removal from head and tail, as well as dynamically changed size when capacity is running out ? 

if so, how? 

please adjust it to have both removeFirst & removeLast to be O(1) operations